gremlin-dotnet/src/Gremlin.Net/Driver/AsyncAutoResetEvent.cs

```diff
@@ -0,0 +1,103 @@
+﻿#region License
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+// The implementation is based on this blog post by Stephen Toub:
```

Comment:
As we are licensing this source code, please note that the source code is original, even if it was inspired by Stephen Toub concepts / ideas.

Reply:
I'm not sure how to express that without stating me explicitly as the author which I would like to avoid. I changed the formulation a bit to make it more clear that the source code was not copied. Let me know if you have any other suggestions.